### PR TITLE
Get rid of deprecated np.asmatrix()

### PR DESCRIPTION
**Pull request by mikiereed**
Per lecture in EE364A on 7/25/23. np.asmatrix() has been long deprecated and should be removed from code

### CODE REVIEW

File: python/ml_estim_incr_signal_data.py [hunk 1]
```diff
@@ -11,12 +11,11 @@
...
+# pass the increasing input through a moving-average filter
 # and add Gaussian noise
 h = np.array([1, -0.85, 0.7, -0.3])
 k = h.size
 yhat = np.convolve(h,xtrue)
 y = yhat[0:-3].reshape(N,1) + np.random.randn(N,1)
 
-xtrue = np.asmatrix(xtrue.reshape(N,1))
-y = np.asmatrix(y)
+xtrue = xtrue.reshape(N,1)
```
Comment (PTNobel) [inline python/ml_estim_incr_signal_data.py]: I wonder if we can get rid of all the reshapes, and just have `h, y, xtrue` all be one dimensional arrays?

File: python/simple_portfolio_data.py [hunk 2]
```diff
@@ -2,10 +2,11 @@
...
+pbar = np.ones((n,1))*.03 + np.r_[np.random.rand(n-1,1), np.zeros((1,1))]*.12
+S = np.random.randn(n, n)
 S = S.T*S
 S = S/max(np.abs(np.diag(S)))*.2
-S[:, -1] = np.zeros((n, 1))
-S[-1, :] = np.zeros((n, 1)).T
-x_unif = np.ones((n, 1))/n; x_unit = np.asmatrix(x_unif)
+S[:, -1] = np.zeros(n)
+S[-1, :] = np.zeros(n).T
+x_unif = np.ones((n, 1))/n
```
Comment (PTNobel) [inline python/simple_portfolio_data.py]: Any reason to make x_unif a (n, 1) instead of (n,) shaped array?

**Review by PTNobel — APPROVED** [review]
Overall this looks perfect! Once you let me know out of channel what changes needs to happen to the solutions we'll merge.

File: python/ml_estim_incr_signal_data.py (hunk 1 above)
Comment (mikiereed) [reply python/ml_estim_incr_signal_data.py]: Good point. It has been updated.
Note that the solution file needs to be updated as well.
See out of channel comments.

File: python/simple_portfolio_data.py (hunk 2 above)
Comment (mikiereed) [reply python/simple_portfolio_data.py]: Nope. Updated.

**Pull request merged**